lambda/audit.py

```diff
@@ -284,7 +311,13 @@ def make_audit_event(
     return locals()
```

Comment:
This should be a named `AuditEvent` dataclass, right? That way it can be typechecked by mypy, and give us the same guarantees as a dictionary at runtime

Reply:
I've changed this to a dataclass. Fixed the logging to work with dataclasses and fixed up the tests.